test/functional/README.md

```diff
@@ -118,3 +118,36 @@ Helpers for script.py
 
 #### [test_framework/blocktools.py](test_framework/blocktools.py)
 Helper functions for creating blocks and transactions.
+
+### Benchmarking with perf
```

Comment:
This documentation is excellent. I think it should live in `/test/README.md`, which is documentation for *running* the functional tests, and already includes hints on logging, attaching a debugger, etc. `/test/functional/README.md` is documentation for *writing* test cases (we could probably add a note to the top of each file explaining that).